apps/cowswap-frontend/src/modules/tokensList/pure/ChainsSelector/index.tsx

```diff
@@ -30,16 +25,21 @@ export function ChainsSelector({
   defaultChainId,
   isLoading,
   // TODO: change the value to 7 after tests
-  itemsToDisplay = 3,
+  visibleNetworkIcons = 3,
 }: ChainsSelectorProps) {
-  const isDisplayMore = chains.length > itemsToDisplay
-
-  const visibleChains = chains.slice(0, itemsToDisplay)
-  const menuChains = chains.slice(itemsToDisplay)
-  const selectedMenuChain = menuChains.find((i) => i.id === defaultChainId)
+  const isMobile = useMediaQuery(Media.upToSmall(false))
+  const isDisplayMore = !isMobile && chains.length > visibleNetworkIcons
+  const visibleChains = isMobile ? chains : chains.slice(0, visibleNetworkIcons)
+  const selectedMenuChain = !isMobile && chains.find((i) => i.id === defaultChainId && !visibleChains.includes(i))
 
   if (isLoading) {
```

Comment:
Could you address this comment please?:
https://github.com/cowprotocol/cowswap/pull/5489#discussion_r1994054529